cypress/e2e/26-parameter-input.hint.cy.ts

```diff
@@ -0,0 +1,47 @@
+import { WorkflowPage, NDV } from '../pages';
+
+const wf = new WorkflowPage();
+const ndv = new NDV();
+
+describe('Parameter input hint', () => {
+	before(() => {
+		cy.resetAll();
+		cy.skipSetup();
+	});
+
+	[
+		{
+			name: 'should display a regular string',
+			pinData: [{ regular_string: 'hello' }],
+			input: '{{ $json.regular_string',
+			output: 'hello',
+		},
+		{
+			name: 'should display `[empty]` for a zero-length string',
+			pinData: [{ empty_string: '' }],
+			input: '{{ $json.empty_string',
+			output: '[empty]',
+		},
+		{
+			name: 'should display ` ` for a single-whitespace string',
+			pinData: [{ single_whitespace_string: ' ' }],
+			input: '{{ $json.single_whitespace_string',
+			output: ' ',
+		},
+	].forEach(({ name, pinData, input, output }) => {
```

Comment:
Only for expediency. e2e tests should be few at the top of the pyramid and units tests at the base, but generally that is not what we have been doing, so I followed the pattern, esp. as e2e is quicker to write and I really want to close this given all the time spent for very little value. But let me know if you want to refactor.

Reply:
hmm it's probably worth taking the time doing this well rather than quick.. especially since these tests don't really add anything here that's not tested by other features..